[18.0][MIG] stock_available_mrp: Migration to 18.0

StefanRijnhart opened the pull request:
Superseding https://github.com/OCA/stock-logistics-availability/pull/38

* Replace product type 'product' by 'consu' + is_storable
* Apply fields.Command where possible
* Rebase to include https://github.com/OCA/stock-logistics-availability/pull/32
* Squash administrative commits somewhat
* Remove unused config parameter
* Improve config settings description in a separate commit so it can be picked into 17 if desired
* Fix 'configuration parameter configured in settings UI is actually unused', also in a separate commit for backporting
* Fix issue where variant specific BoM was not preferred, by @dcatarino 
* Fix issue with quantities of kit products, by @ellbristow

sbidoul committed:
[MOV] move addons out of __unported__ (they remain not installable)

sbidoul committed:
[MOV] move addons out of __unported__ (they remain not installable)

lmignon committed:
[FIX] Browse mrp.bom as current user
sudo is not required since mrp.bom are readable to groups with access to the qty_x fields on a product. Moreover using sudo to retrive the bom will ignore the company_id defined on the bom

lmignon committed:
[FIX] Browse mrp.bom as current user
sudo is not required since mrp.bom are readable to groups with access to the qty_x fields on a product. Moreover using sudo to retrive the bom will ignore the company_id defined on the bom

Commit:
[MIGR] stock_available_mrp: migrate to v8

Compute potential quantities for both product templates and variants. To keep the code simple, only the biggest potential of any single variant is accounted for in the template's potential.
Take all levels of phantom BoM into account, respects validity dates etc. thanks to the use of the standard method _bom_explode, as suggested by @gdgellatly in https://github.com/OCA/stock-logistics-warehouse/pull/5#issuecomment-66902191
Improve tests, rewritten in python.
Adhere to new file/manifest/README conventions.
Simplify copyright headers

Commit:
[MIGR] stock_available_mrp: migrate to v8

Compute potential quantities for both product templates and variants. To keep the code simple, only the biggest potential of any single variant is accounted for in the template's potential.
Take all levels of phantom BoM into account, respects validity dates etc. thanks to the use of the standard method _bom_explode, as suggested by @gdgellatly in https://github.com/OCA/stock-logistics-warehouse/pull/5#issuecomment-66902191
Improve tests, rewritten in python.
Adhere to new file/manifest/README conventions.
Simplify copyright headers

Commit:
[FIX] fix multi-company test

Record rules used to not be checked on stock quants, but now they are since Odoo's commit 2fd14db57433d08ab368a7e18d18ae8d49dafab1 (https://github.com/odoo/odoo/commit/2fd14db).
In our test we changed the company of the products and BoMs but we neglected that the stock was not attached to the right company, and that made the test fail.
To fix that, make the test inventory for the right company.
Since there is a little inconsistency in the demo data with a negative quantity of an unrelated product, use the `partial` filter for the inventories instead of the `none` filter, so that no wrong inventory lines are added automatically.

Commit:
[FIX] fix multi-company test

Record rules used to not be checked on stock quants, but now they are since Odoo's commit 2fd14db57433d08ab368a7e18d18ae8d49dafab1 (https://github.com/odoo/odoo/commit/2fd14db).
In our test we changed the company of the products and BoMs but we neglected that the stock was not attached to the right company, and that made the test fail.
To fix that, make the test inventory for the right company.
Since there is a little inconsistency in the demo data with a negative quantity of an unrelated product, use the `partial` filter for the inventories instead of the `none` filter, so that no wrong inventory lines are added automatically.

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
stock_available / stock_available_mrp : add possibility to choose which field use to compute potential

Commit:
stock_available / stock_available_mrp : add possibility to choose which field use to compute potential

Commit:
stock_available_mrp: fix BOM qty (multi-units, efficiency)

Commit:
stock_available_mrp: fix BOM qty (multi-units, efficiency)

Commit:
[FIX+IMP] stock_available_mrp: fix recursive potential_qty in list view + small improvements

Commit:
[FIX+IMP] stock_available_mrp: fix recursive potential_qty in list view + small improvements

Commit:
V9 migration modifications.

    * mrp_bom.name has been deleted.
    * mrp_bom_line.type moved to mrp_bom.type.
    * Fix missing group_mrp_user issue.
    * Change versions

Commit:
V9 migration modifications.

    * mrp_bom.name has been deleted.
    * mrp_bom_line.type moved to mrp_bom.type.
    * Fix missing group_mrp_user issue.
    * Change versions

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[10.0][MIG] port stock_available_mrp to 10.0

Commit:
[10.0][MIG] port stock_available_mrp to 10.0

rousseldenis committed:
[IMP] stock_available_mrp: readability + change compute bom_id + use recordset + Condition never statisfied

rousseldenis committed:
[IMP] stock_available_mrp: readability + change compute bom_id + use recordset + Condition never statisfied

Commit:
[ADD] include new improvments

Commit:
[ADD] include new improvments

Commit:
[FIX] stock_available_mrp: remove duplicate button + fix calculation of immediately usable qty

Commit:
[FIX] stock_available_mrp: remove duplicate button + fix calculation of immediately usable qty

Commit:
[11.0][MIG] port stock_available_mrp to 11.0

Commit:
[11.0][MIG] port stock_available_mrp to 11.0

florian-dacosta committed:
Migrate stock_available_mrp to v12

florian-dacosta committed:
Migrate stock_available_mrp to v12

florian-dacosta committed:
Make stock_available_mrp compatible with other modules like stock_available_immediately

florian-dacosta committed:
Make stock_available_mrp compatible with other modules like stock_available_immediately

pedrobaeza committed:
[FIX] stock_available_mrp: Fix RST formatting + generate README

pedrobaeza committed:
[FIX] stock_available_mrp: Fix RST formatting + generate README

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (7 of 7 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_available_mrp/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (7 of 7 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_available_mrp/zh_CN/

Commit:
Improve performance of stock available of BoMs

Use an optimized method to explode the BoM. The explode methods
include data and a cycle check that we don't need here. Besides,
it calls '_bom_find' on every bom line of the graph, which generates
thousands of SELECT queries on large BoMs.

guewen committed:
Improve performance of stock available of BoMs

Use an optimized method to explode the BoM. The explode methods
include data and a cycle check that we don't need here. Besides,
it calls '_bom_find' on every bom line of the graph, which generates
thousands of SELECT queries on large BoMs.

guewen committed:
Use Savepoint in stock_available_mrp tests

Commit:
Use Savepoint in stock_available_mrp tests

ypapouin committed:
Translated using Weblate (French)

Currently translated at 100.0% (7 of 7 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_available_mrp/fr/

ypapouin committed:
Translated using Weblate (French)

Currently translated at 100.0% (7 of 7 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_available_mrp/fr/

chienandalu committed:
[FIX] stock_available_mrp: upstream compatibility

After this change https://github.com/odoo/odoo/commit/3d34d58388ab362bf8e69cc2e75adbdfda3331b2
it isn't possible to have an storable kit product anymore. The main
issue is that we have to drop some uses cases that this module
considered for this kind of situations.

chienandalu committed:
[FIX] stock_available_mrp: upstream compatibility

After this change https://github.com/odoo/odoo/commit/3d34d58388ab362bf8e69cc2e75adbdfda3331b2
it isn't possible to have an storable kit product anymore. The main
issue is that we have to drop some uses cases that this module
considered for this kind of situations.

chienandalu committed:
[FIX] stock_available_mrp: don't force type

In Odoo change odoo/odoo@3d34d58 the products with Kit BoMs couldn't
longer be other than `consu`. This was reverted in odoo/odoo@6c4623a
and some products are raising errors when forcing the type

chienandalu committed:
[FIX] stock_available_mrp: don't force type

In Odoo change odoo/odoo@3d34d58 the products with Kit BoMs couldn't
longer be other than `consu`. This was reverted in odoo/odoo@6c4623a
and some products are raising errors when forcing the type

victoralmau committed:
[IMP] stock_available_mrp: black, isort, prettier

victoralmau committed:
[IMP] stock_available_mrp: black, isort, prettier

victoralmau committed:
[MIG] stock_available_mrp: Migration to 13.0

victoralmau committed:
[MIG] stock_available_mrp: Migration to 13.0

florian-dacosta committed:
Migrate stock_available_mrp to v14

florian-dacosta committed:
Migrate stock_available_mrp to v14

OCA-git-bot committed:
stock_available_mrp 14.0.1.0.1

OCA-git-bot committed:
stock_available_mrp 14.0.1.0.1

sbejaoui committed:
[FIX] - local variable referenced before assignment

sbejaoui committed:
[FIX] - local variable referenced before assignment

Commit:
[FIX] adapt back to V14

Commit:
[FIX] adapt back to V14

Commit:
[14.0] stock_available_mrp: improve performance by using standard field

Commit:
[14.0] stock_available_mrp: improve performance by using standard field

Commit:
[14.0] stock_available: improve search performance

Commit:
[14.0] stock_available: improve search performance

oca-travis committed:
[UPD] Update stock_available_mrp.pot

oca-travis committed:
[UPD] Update stock_available_mrp.pot

OCA-git-bot committed:
stock_available_mrp 14.0.1.0.3

OCA-git-bot committed:
stock_available_mrp 14.0.1.0.3

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-14.0/stock-logistics-warehouse-14.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-14-0/stock-logistics-warehouse-14-0-stock_available_mrp/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-14.0/stock-logistics-warehouse-14.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-14-0/stock-logistics-warehouse-14-0-stock_available_mrp/

JoanMForgeFlow committed:
[15.0][IMP] stock_available_mrp: black, isort, prettier

JoanMForgeFlow committed:
[15.0][IMP] stock_available_mrp: black, isort, prettier

JoanMForgeFlow committed:
[15.0][MIG] stock_available_mrp: Migration to 15.0

JoanMForgeFlow committed:
[15.0][MIG] stock_available_mrp: Migration to 15.0

victoralmau committed:
[FIX] stock_available_mrp: Set correctlly immediately_usable_qty value.
TT27873

victoralmau committed:
[FIX] stock_available_mrp: Set correctlly immediately_usable_qty value.
TT27873

rousseldenis committed:
[TESTS] stock_available_mrp: Add test for phantom bom

rousseldenis committed:
[TESTS] stock_available_mrp: Add test for phantom bom

OCA-git-bot committed:
stock_available_mrp 15.0.1.0.1

OCA-git-bot committed:
stock_available_mrp 15.0.1.0.1

jabelchi committed:
Added translation using Weblate (Catalan)

jabelchi committed:
Added translation using Weblate (Catalan)

jabelchi committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (5 of 5 strings)

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_available_mrp/ca/

jabelchi committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (5 of 5 strings)

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_available_mrp/ca/

Commit:
[16.0][MIG] stock_available_mrp: Migration to 16.0

khoivo1601 committed:
[16.0][MIG] stock_available_mrp: Migration to 16.0

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (10 of 10 strings)

Translation: stock-logistics-availability-16.0/stock-logistics-availability-16.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-availability-16-0/stock-logistics-availability-16-0-stock_available_mrp/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (10 of 10 strings)

Translation: stock-logistics-availability-16.0/stock-logistics-availability-16.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-availability-16-0/stock-logistics-availability-16-0-stock_available_mrp/es/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-availability-16.0/stock-logistics-availability-16.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-availability-16-0/stock-logistics-availability-16-0-stock_available_mrp/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-availability-16.0/stock-logistics-availability-16.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-availability-16-0/stock-logistics-availability-16-0-stock_available_mrp/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (2 of 2 strings)

Translation: stock-logistics-availability-16.0/stock-logistics-availability-16.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-availability-16-0/stock-logistics-availability-16-0-stock_available_mrp/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (2 of 2 strings)

Translation: stock-logistics-availability-16.0/stock-logistics-availability-16.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-availability-16-0/stock-logistics-availability-16-0-stock_available_mrp/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: stock-logistics-availability-16.0/stock-logistics-availability-16.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-availability-16-0/stock-logistics-availability-16-0-stock_available_mrp/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: stock-logistics-availability-16.0/stock-logistics-availability-16.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-availability-16-0/stock-logistics-availability-16-0-stock_available_mrp/it/

bizzappdev committed:
[IMP] stock_available_mrp: pre-commit auto fixes

bizzappdev committed:
[IMP] stock_available_mrp: pre-commit auto fixes

bizzappdev committed:
[MIG] stock_available_mrp: Migration to 17.0

bizzappdev committed:
[MIG] stock_available_mrp: Migration to 17.0

Commit:
[FIX] stock_available_mrp: increment potencial_qty to immediately_usable_qty instead of replacing it

Commit:
[FIX] stock_available_mrp: increment potencial_qty to immediately_usable_qty instead of replacing it

danielduqma committed:
[FIX] stock_available_mrp: variant BoMs

When a BoM is linked just to one variant through field `product_id`,
it is taken in account when computing stock of the rest of variants
of the same template.

danielduqma committed:
[FIX] stock_available_mrp: variant BoMs

When a BoM is linked just to one variant through field `product_id`,
it is taken in account when computing stock of the rest of variants
of the same template.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: stock-logistics-availability-17.0/stock-logistics-availability-17.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-availability-17-0/stock-logistics-availability-17-0-stock_available_mrp/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: stock-logistics-availability-17.0/stock-logistics-availability-17.0-stock_available_mrp
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-availability-17-0/stock-logistics-availability-17-0-stock_available_mrp/it/

StefanRijnhart committed:
[MIG] stock_available_mrp: Migration to 18.0

StefanRijnhart committed:
[MIG] stock_available_mrp: Migration to 18.0

StefanRijnhart committed:
[IMP] stock_available_mrp: improve configuration setting description

StefanRijnhart committed:
[IMP] stock_available_mrp: improve configuration setting description

StefanRijnhart committed:
[FIX] stock_available_mrp: clear up confusion about configuration parameters

The configuration parameter in which the field to use was stored was actually not
used. Instead, another configuration parameter, for which a default value was
provided in xml data was used.

StefanRijnhart committed:
[FIX] stock_available_mrp: clear up confusion about configuration parameters

The configuration parameter in which the field to use was stored was actually not
used. Instead, another configuration parameter, for which a default value was
provided in xml data was used.

StefanRijnhart commented:
/ocabot migration stock_available_mrp

StefanRijnhart committed:
[IMP] stock_available_mrp: pre-commit auto fixes

StefanRijnhart committed:
[IMP] stock_available_mrp: pre-commit auto fixes

ellbristow commented on stock_available_mrp/models/product_product.py:
```diff
@@ -0,0 +1,188 @@
+                )
+
+            res[product.id]["potential_qty"] = potential_qty
+            res[product.id]["immediately_usable_qty"] += potential_qty
```
In the case of Kits this is wrong. Odoo already includes the buildable quantity upstream, so this modification is adding additional stock (double counting the Available stock). By returning this to = and limiting this to Kits, both kits and manufactured show the correct Available qty.

```suggestion
            if bom_id.type == "phantom":
                res[product.id]["immediately_usable_qty"] = potential_qty
```

StefanRijnhart replied:
Apologies for the late response and thank you very much for suggesting the fix. Added in a separate commit.

ellbristow reviewed (changes requested):
This caused problems for one of our customers who send the immediately usable quantity to Magento via EDI and it was sending the wrong stock figures.

dcatarino committed:
[FIX] stock_available_mrp: prefer product-specific BoMs

Use variant-specific BoMs when computing potential stock, including nested phantom BoMs, instead of selecting the first template BoM.

StefanRijnhart committed:
[FIX] stock_available_mrp: Odoo already provides buildable quantities for kit products

Co-authored-by: Elliott Bristow <elliott@glo.systems>

flotho reviewed (approved):
Code review, LGTM
Thanks for this
